repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -455,9 +455,11 @@ public EntityMutationResponse mapAttributesAndClassifications(EntityMutationCont
                     setSystemAttributesToEntity(vertex,updatedEntity);
                     resp.addEntity(updateType, constructHeader(updatedEntity, vertex, entityType.getAllAttributes()));
 
-                    Set<AtlasEdge> inOutEdges = getNewCreatedInputOutputEdges(guid);
+                    Set<AtlasEdge> inOutEdges = new HashSet<>();
+                    inOutEdges.addAll(getNewCreatedInputOutputEdges(guid));
+                    inOutEdges.addAll(getRestoredInputOutputEdges(vertex));
```

Comment:
Call getRestoredInputOutputEdges only when it is restored vertex. It will add unnecessary overhead when it is not a  normal vertex.
 You can move the condition of `isRestoreEntity` outside the if block and reuse it

Reply:
addressed